### DISCUSSION TITLE
Enable/Disable API console for specific API's (Feature Request)

### DISCUSSION THREAD

**Discussion by LauraXiulan** [Azure/api-management-developer-portal#2620] [Feature request]
Currently when toggling the Enable API console button, it is turned on and off for all API's. We would like to be able to turn off the Try it console for some of the API's we provide, but not for all of them. Especially for our endpoints that return Excel/CSV files. For those the Try it console is currently unusable because the response is printed out in the screen instead of downloaded.

We are using the managed developer portal and are not planning to self-host it.

**Comment by malincrist** [comment]
Hello @LauraXiulan , thank you for the feature request. We do not have such thing in plan yet.
However, I believe you can achieve this behavior by implementing a custom widget, where you can manipulate the `try it` button based on your specific needs.
You can find more info [here](https://learn.microsoft.com/en-us/azure/api-management/developer-portal-extend-custom-functionality#create-and-upload-custom-widget).
